Source/WebKit/WebKit.xcodeproj/project.pbxproj

```diff
@@ -16663,6 +16666,26 @@
 			shellPath = /bin/sh;
 			shellScript = "echo \"Preprocessing sandbox\"\nScripts/generate-derived-sources.sh sandbox-profiles-ios\nmkdir -p ${DSTROOT}/${INSTALL_PATH}\ncp ${BUILT_PRODUCTS_DIR}/DerivedSources/WebKit/com.apple.WebKit.adattributiond.sb ${DSTROOT}/${INSTALL_PATH}\ncp ${BUILT_PRODUCTS_DIR}/DerivedSources/WebKit/com.apple.WebKit.GPU.sb ${DSTROOT}/${INSTALL_PATH}\ncp ${BUILT_PRODUCTS_DIR}/DerivedSources/WebKit/com.apple.WebKit.Networking.sb ${DSTROOT}/${INSTALL_PATH}\ncp ${BUILT_PRODUCTS_DIR}/DerivedSources/WebKit/com.apple.WebKit.WebAuthn.sb ${DSTROOT}/${INSTALL_PATH}\ncp ${BUILT_PRODUCTS_DIR}/DerivedSources/WebKit/com.apple.WebKit.WebContent.sb ${DSTROOT}/${INSTALL_PATH}\n";
 		};
+		EBE4D2AD28A2F3BD00C0FAE7 /* Copy Logging Preferences */ = {
+			isa = PBXShellScriptBuildPhase;
+			buildActionMask = 8;
+			files = (
+			);
+			inputFileListPaths = (
+			);
+			inputPaths = (
+				"$(SRCROOT)/Resources/LoggingPreferences.plist",
+			);
+			name = "Copy Logging Preferences";
+			outputFileListPaths = (
+			);
+			outputPaths = (
+				"$(DSTROOT)/System/Library/Preferences/Logging/Subsystems/com.apple.WebKit.plist",
+			);
+			runOnlyForDeploymentPostprocessing = 1;
+			shellPath = /bin/sh;
+			shellScript = "set -e\n\nif [[ \"${WK_PLATFORM_NAME}\" == maccatalyst || \"${WK_PLATFORM_NAME}\" == iosmac ]]; then\n    exit 0\nfi\n\ncp \"${SCRIPT_INPUT_FILE_0}\" \"${SCRIPT_OUTPUT_FILE_0}\"\nplutil -convert binary1 \"${SCRIPT_OUTPUT_FILE_0}\"\n";
```

Comment:
Why no signposts for MacCatalyst apps?

If we can use signposts on MacCatalyst builds, can we use a plain "copy file" build phase instead of a custom script here?

Reply:
The build phase is to copy a preference plist to `/System/Library/Preferences/Logging/Subsystems/com.apple.WebKit.plist`. This changes all signposts in the `com.apple.WebKit` subsystem to be persisted by default (rather than in-memory only). Lots of subsystems do this, the reason being that in-memory signpost storage makes it so that you see only an arbitrary number of signposts in your artrace, depending on the state of the in-memory log buffer at the time of the trace. That makes it harder to get started with tracing.

We skip copying the file for `maccatalyst` and `iosmac` because as far as I understand, `maccatalyst` and `iosmac` do not have their own logd. Instead they just use the logd that's running as part of the base macOS system.

Reply:
Makes sense.  r=me

Reply:
I would prefer a native Copy Files build phase, with EXCLUDED_SOURCE_FILE_NAMES set in WebKit.xcconfig to avoid copying LoggingPreferences.plist for catalyst builds. It avoids us having to ensure the input/output paths are declared correctly, and is a bit less overhead.

Reply:
@emw-apple Is there a way to rename a file in a Copy Files build phase? I'm copying these files:

```
$(SRCROOT)/Resources/Signposts/LoggingPreferences.plist
$(SRCROOT)/Resources/Signposts/SystemTracePoints.plist
```

To these destinations:

```
$(DSTROOT)/System/Library/Preferences/Logging/Subsystems/com.apple.WebKit.plist
$(DSTROOT)/AppleInternal/Library/Ariadne/Plists/WebKit.plist
```

I could rename the source filenames to match the destination filenames, but the filenames `com.apple.WebKit.plist` and `WebKit.plist` are awfully generic, and seem like they would pollute Xcode's Open Quickly, etc.

Reply:
You could do it using a build rule (and APPLY_RULES_IN_COPY_FILES=YES, which WebKit should already have enabled), though you'd have to write out a shell script to do the copying, at which point you might as well use the original script phase 😄

That said, I think it might actually be better to keep the names the same, since it would allow someone to see the file in build products (or installed in a root) and more easily figure out where that file comes from in the project. The `Source/WebKit/Resources/Signposts` path (which Xcode shows in Open Quickly) makes up for some of that lost context.

Reply:
I don't anticipate that these files will be changed often (or at all). Given that, I think for now I want to err on the side of not polluting people's autocomplete for `WebKit` in Xcode. But if you feel really strongly then let me know and I will file a follow up patch to replace this build phase with a copy files script.

Reply:
I don't feel strongly, LGTM :)